internals/gcp/service_creator.go

```diff
@@ -0,0 +1,63 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/secrethub/secrethub-go/internals/api"
+	"google.golang.org/api/cloudkms/v1"
+	"google.golang.org/api/option"
+)
+
+// CredentialCreator is an implementation of the secrethub.Verifier and secrethub.Encrypter interface that can be used
+// to create an GCP service account.
+type CredentialCreator struct {
+	keyResourceID       string
+	serviceAccountEmail string
+
+	encryptFunc func(name string, plaintext string) (*cloudkms.EncryptResponse, error)
+}
+
+// NewCredentialCreator returns a CredentialCreator that uses the provided GCP KMS key and Service Account Email to create a new credential.
+// The GCP client is configured with the optionally provided option.ClientOption.
+func NewCredentialCreator(serviceAccountEmail, keyResourceID string, gcpOptions ...option.ClientOption) (*CredentialCreator, map[string]string, error) {
```

Comment:
Is `resource ID` what it's called? I've also seen `self link` and `resource name`. Not sure 'what is wisdom here'.

Reply:
But anyway: let's pick one and use that everywhere (in the CLI and Terraform provider too).

Reply:
It's what Google calls it
![image](https://user-images.githubusercontent.com/1965218/83645349-ab01b900-a5b2-11ea-9ce3-f31c6ed023e6.png)